### PR TITLE
grpc: Add GetAccountExtendedKey command

### LINKED ISSUE #1196: Add AccountExtendedKey gRPC command
To allow Decrediton user's to request a given accounts' masterpubkey for use with the recently implemented watch-only wallets in Decrediton.

This should be added into the WalletService.
The AccountExtendedKeyRequest should accept an option for which account's pubkey to return.  

The AccountExtendedKeyResponse should just contain the pubkey in the form of `bytes master_pubkey`

An error should be returned if the request contains an account number that is beyond the current account index since the xpub for that account hadn't been derived and as such would require the wallet to be unlocked.

### PR DESCRIPTION
**Pull request by vctt94**
closes #1196

### CODE REVIEW

File: rpc/documentation/api.md [hunk 1]
```diff
@@ -627,6 +627,23 @@ but not all sidechain blocks may be known by the wallet.
 
 ___
 
+#### `GetAccountExtendedKey`
+
+The `GetAccountExtendedKey` method queries the wallet for an account pubkey. If no account is provided, then it returns the master pubkey.
+
+**Request:** `GetAccountExtendedKeyRequest`
+
+- `string account`: The account name to retrieve the pubkey.
```
Comment (jrick) [inline rpc/documentation/api.md]: the grpc api refers to accounts as a uint32.  This method must remain consistent.

File: rpc/documentation/api.md [hunk 2]
```diff
@@ -627,6 +627,23 @@ but not all sidechain blocks may be known by the wallet.
 
 ___
 
+#### `GetAccountExtendedKey`
+
+The `GetAccountExtendedKey` method queries the wallet for an account pubkey. If no account is provided, then it returns the master pubkey.
```
Comment (jrick) [inline rpc/documentation/api.md]: if the account can't be found, this must error.

File: rpc/documentation/api.md [hunk 3]
```diff
@@ -627,6 +627,23 @@ but not all sidechain blocks may be known by the wallet.
 
 ___
 
+#### `GetAccountExtendedKey`
```
Comment (jrick) [inline rpc/documentation/api.md]: Name this GetAccountExtendedPubKey, simply Key is ambiguous

**Comment by alexlyp** [comment]
Needs a rpc version bump as well.

**Comment by vctt94** [comment]
Fixed.

**Comment by alexlyp** [comment]
mmm no, you need to bump the version at the top of server.go

File: rpc/rpcserver/server.go [hunk 4]
```diff
@@ -941,6 +941,18 @@ func (s *walletServer) ConstructTransaction(ctx context.Context, req *pb.Constru
 	return res, nil
 }
 
+func (s *walletServer) GetAccountExtendedPubKey(ctx context.Context, req *pb.GetAccountExtendedPubKeyRequest) (*pb.GetAccountExtendedPubKeyResponse, error) {
+
```
Comment (jrick) [inline rpc/rpcserver/server.go]: please remove this newline

Comment (vctt94) [reply rpc/rpcserver/server.go]: fixed

File: rpc/api.proto [hunk 5]
```diff
@@ -927,6 +928,14 @@ message CommittedTicketsRequest {
 	repeated bytes tickets = 1;
 }
 
+message GetAccountExtendedPubKeyRequest {
+	uint32 account_number = 1;
+}
+
+message GetAccountExtendedPubKeyResponse {
+	string AccountExtendedPubKey = 1;
```
Comment (jrick) [inline rpc/api.proto]: this name should not be camel case

File: rpc/documentation/api.md [hunk 6]
```diff
@@ -627,6 +627,23 @@ but not all sidechain blocks may be known by the wallet.
...
+
+The `GetAccountExtendedPubKey` method queries the wallet for an account pubkey. If the provided account number is beyond the current account index, then it returns an error.
+
+**Request:** `GetAccountExtendedPubKeyRequest`
+
+- `uint32 account_number`: The number of the account to retrieve the pubkey.
+
+**Response:** `GetAccountExtendedPubKeyResponse`
+
+- `string extendedkey`: The account's extended key.
```
Comment (jrick) [inline rpc/documentation/api.md]: name doesn't match actuality

**Comment by vctt94** [comment]
fixed

File: rpc/documentation/api.md [hunk 7]
```diff
@@ -627,6 +627,23 @@ but not all sidechain blocks may be known by the wallet.
 
 ___
 
+#### `GetAccountExtendedPubKey`
+
+The `GetAccountExtendedPubKey` method queries the wallet for an account pubkey. If the provided account number is beyond the current account index, then it returns an error.
```
Comment (jrick) [inline rpc/documentation/api.md]: keep the documentation wrapped to 80 columns please

We also don't need to document in the description that a nonexistent account will cause an error, that's expected (and already documented in the expected errors section below).

**Comment by vctt94** [comment]
fixed

**Pull request merged**